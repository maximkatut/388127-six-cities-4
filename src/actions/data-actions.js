import {ActionType} from './types';

export const ActionCreator = {
  loadOffers: (offers) => {
    return {
      type: ActionType.LOAD_OFFERS,
      payload: offers,
    };
  },
  loadReviews: (reviews) => {
    return {
      type: ActionType.LOAD_REVIEWS,
      payload: reviews,
    };
  },
  loadOffersNearby: (offersNearby) => {
    return {
      type: ActionType.LOAD_OFFERS_NEARBY,
      payload: offersNearby,
    };
  },
};
