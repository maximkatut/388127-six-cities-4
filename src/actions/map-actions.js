import {ActionType} from './types';

export const ActionCreator = {
  changeCardIdOnHover: (cardIdOnHover) => {
    return {
      type: ActionType.CHANGE_CARD_ON_HOVER,
      payload: cardIdOnHover
    };
  }
};
