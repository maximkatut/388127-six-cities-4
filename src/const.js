export const OfferTypes = [
  `Apartment`, `Room`, `House`, `Hotel`
];

// MAP CONSTS
export const MAP_ZOOM = 12;
export const MAP_ICON_SIZE = [27, 39];
export const MAP_ICON_URL = `img/pin.svg`;
