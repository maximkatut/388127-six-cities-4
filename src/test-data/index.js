export const offers = [
  {
    appliences: [`Wi-Fi`, `Microwave`],
    bedrooms: 3,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    guests: 2,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: true
    },
    id: `1`,
    imagesGallery: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    mainImage: `img/apartment-01.jpg`,
    offerType: `House`,
    premium: true,
    price: 200,
    rate: 5,
    title: `Nice, cozy, warm big bed house`
  },
  {
    appliences: [`Wi-Fi`, `Microwave`],
    bedrooms: 4,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: [`An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    guests: 3,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Anton`,
      super: false
    },
    id: `2`,
    imagesGallery: [
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    mainImage: `img/apartment-02.jpg`,
    offerType: `Apartment`,
    premium: true,
    price: 300,
    rate: 4,
    title: `Nice, cozy, warm big bed apartment`
  }
];

