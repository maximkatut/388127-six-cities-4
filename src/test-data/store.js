import {cities, offers} from '.';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

export const store = mockStore({
  activeCityName: cities[0].name,
  offers,
  cities
});
