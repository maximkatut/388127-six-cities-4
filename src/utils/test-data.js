export const offerCards = [
  {
    id: `1`,
    imageSrc: `img/apartment-02.jpg`,
    offerType: `Room`,
    premium: false,
    price: 200,
    rate: 5,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `2`,
    imageSrc: `img/apartment-03.jpg`,
    offerType: `House`,
    premium: true,
    price: 300,
    rate: 4,
    title: `Nice, cozy, warm big bed apartment`
  }
];
