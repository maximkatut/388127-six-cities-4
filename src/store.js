import {applyMiddleware, createStore} from 'redux';
import {composeWithDevTools} from "redux-devtools-extension";
import thunk from 'redux-thunk';

import {ActionCreator as DataActionCreator} from './actions/data-actions';
import {ActionCreator as UserActionCreator} from './actions/user-actions';
import {createAPI} from './api';
import {AppRoute} from './const';
import history from './history';
import rootReducer from './reducers';
import {Operation as DataOperation} from './reducers/data/data-reducer';
import {AuthorizationStatus, Operation as UserOperation} from './reducers/user/user-reducer';

const onUnauthorized = () => {
  store.dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH));
  store.dispatch(UserActionCreator.getUser({}));
  if (history.location.pathname === AppRoute.FAVORITES) {
    history.push(AppRoute.LOGIN);
  }
};

const onBadRequest = (err) => {
  store.dispatch(DataActionCreator.catchError(true, err));
};

const api = createAPI(onUnauthorized, onBadRequest);

export const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);

store.dispatch(DataOperation.loadOffers());
store.dispatch(UserOperation.checkAuth());
