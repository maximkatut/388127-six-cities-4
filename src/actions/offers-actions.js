import {ActionType} from './types';
import {offers} from '../mocks/offers';
import {getOffersByCity, getOffersBySort} from '../utils';

export const ActionCreator = {
  changeCity: (cityName) => {
    return {
      type: ActionType.CHANGE_CITY,
      payload: cityName
    };
  },
  getOffers: (cityName) => {
    const filteredOffers = getOffersByCity(cityName, offers);
    return {
      type: ActionType.GET_OFFERS,
      payload: filteredOffers
    };
  },
  changeSortType: (sortType, offersByCity) => {
    const sortedOffers = getOffersBySort(sortType, offersByCity);
    return {
      type: ActionType.CHANGE_SORT_TYPE,
      payload: sortedOffers
    };
  },
  changeActiveOffer: (activeOffer) => {
    return {
      type: ActionType.CHANGE_ACTIVE_OFFER,
      payload: activeOffer
    };
  }
};
