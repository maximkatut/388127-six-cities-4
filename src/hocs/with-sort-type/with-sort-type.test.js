import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

import {store} from '../../test-data/store';
import {ActionCreator} from '../../actions/offers-actions';
import {SortType} from '../../const.js';

import withSortMenu from './with-sort-type';

const MockComponent = (props) => {
  const {onSortClick} = props;

  return (
    <div>
      <li onClick={() => {
        onSortClick(SortType.POPULAR);
      }}></li>
    </div>
  );
};

MockComponent.propTypes = {
  onSortClick: PropTypes.func.isRequired
};

const MockComponentWrapped = withSortMenu(MockComponent);

describe(`withSortMenu`, () => {
  let component;

  store.dispatch = jest.fn();

  beforeEach(() => {
    component = renderer.create((
      <Provider store={store}>
        <MockComponentWrapped/>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    });
  });

  it(`withSortMenu is rendered correctly`, () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`withSortMenu should dispatch an action on sort-menu dropdown click`, () => {
    renderer.act(() => {
      component.root.findAllByType(`li`)[0].props.onClick();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
        ActionCreator.changeSortType(SortType.POPULAR)
    );
  });
});
