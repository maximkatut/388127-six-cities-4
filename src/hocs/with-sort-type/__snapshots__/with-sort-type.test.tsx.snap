// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSortMenu withSortMenu is rendered correctly 1`] = `
<div>
  <li
    onClick={[Function]}
  />
</div>
`;
